samza-api/src/main/java/org/apache/samza/container/grouper/stream/GrouperContext.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.container.grouper.stream;
+
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.List;
+import java.util.Set;
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.container.TaskName;
+import org.apache.samza.runtime.LocationId;
+import org.apache.samza.system.SystemStreamPartition;
+
+/**
+ * A Wrapper class that holds the necessary historical metadata of the samza job which is used
+ * by the {@link org.apache.samza.container.grouper.stream.SystemStreamPartitionGrouper}
+ * to generate optimal task assignments.
+ */
+@InterfaceStability.Evolving
+public class GrouperContext {
+  private Map<String, LocationId> processorLocality;
+  private Map<TaskName, LocationId> taskLocality;
+  private Map<TaskName, Set<SystemStreamPartition>> previousTaskToSSPAssignment;
+  private Map<TaskName, String> previousTaskToContainerAssignment;
```

Comment:
```
Please ad these comments: 
private Map<TaskName, LocationId> taskLocality; // Map of task-name to location ID
private Map<TaskName, Set<SystemStreamPartition>> previousTaskToSSPAssignment; // Map of task-name to the set of input SSPs for each task
private Map<TaskName, String> previousTaskToContainerAssignment; // Map of task-name to the container-name on which the task is hosted.
private Map<String, LocationId> processorLocality; // <Meaning goes here> 
```

Reply:
Also what does ```processorLocality``` mean for YARN deploys?

Reply:
1. Those comments are already added in PR 790. This file will not be checked-in as a part of this PR(I added a comment to indicate that). I added this file here, since PR 790 is in review. Apologies for any confusion.
2. Processor locality for a processor denotes the previous host in which a particular processor ran in YARN.